### PR TITLE
Story 3681, Task 4897 epics public preview

### PR DESCRIPTION
**Pull request by stuart-bass-bjss**

### CODE REVIEW

File: tests/NHSD.BuyingCatalogue.Solutions.Persistence.DatabaseTests/SolutionEpicRepositoryTests.cs [hunk 1]
```diff
@@ -54,71 +45,186 @@ await SolutionEntityBuilder.Create()
...
-            
         }
 
         [Test]
-        public void ShouldThrowIfEpicRequestIsNull()
+        public async Task UpdateSolutionWithOneEpicAsync()
         {
-            Assert.ThrowsAsync<ArgumentNullException>(() =>
-                _solutionEpicRepository.UpdateSolutionEpicAsync(Solution1Id, null, new CancellationToken()));
+
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Solutions.Persistence.DatabaseTests/SolutionEpicRepositoryTests.cs]: whitespace

File: tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/Solutions/SolutionGetByIdTests.cs [hunk 2]
```diff
@@ -571,5 +579,95 @@ public void ShouldThrowWhenSolutionNotPresent()
 
             _context.MockSolutionRepository.Verify(r => r.ByIdAsync("Sln1", It.IsAny<CancellationToken>()), Times.Once());
         }
+
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/Solutions/SolutionGetByIdTests.cs]: Extra new line?

File: tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/Solutions/SolutionGetByIdTests.cs [hunk 3]
```diff
@@ -571,5 +579,95 @@ public void ShouldThrowWhenSolutionNotPresent()
...
+                e.CapabilityId == capabilities1.CapabilityId &&
+                e.EpicName == "Epic 4" &&
+                e.EpicCompliancyLevel == "MAY" &&
+                e.IsMet == false);
+
+            _context.MockSolutionCapabilityRepository
+                .Setup(r => r.ListSolutionCapabilitiesAsync("Sln1", It.IsAny<CancellationToken>())).ReturnsAsync(new[] { capabilities1, capabilities2 });
+
+            _context.MockSolutionEpicRepository
+                .Setup(r => r.ListSolutionEpicsAsync("Sln1", It.IsAny<CancellationToken>())).ReturnsAsync(new[] { epic1, epic2,epic3,epic4 });
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/Solutions/SolutionGetByIdTests.cs]: Spacing between variable names e.g. epic1, epic2, epic3, epic4

File: tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/Solutions/SolutionGetByIdTests.cs [hunk 4]
```diff
@@ -571,5 +579,95 @@ public void ShouldThrowWhenSolutionNotPresent()
...
+                                                                                                   e.IsMet == false &&
+                                                                                                   e.EpicCompliancyLevel == "MUST"),
+                                                              Mock.Of<IClaimedCapabilityEpic>( e=> e.EpicId == "C1E4" &&
+                                                                                                   e.EpicName == "Epic 4" &&
+                                                                                                   e.IsMet == false &&
+                                                                                                   e.EpicCompliancyLevel == "MAY")
+                                                          }),
+                        Mock.Of<IClaimedCapability>(cc => cc.Name == "cap2")
+                    }, config => config.ComparingByMembers<IClaimedCapability>());
+
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/Solutions/SolutionGetByIdTests.cs]: Extra new line?

File: tests/NHSD.BuyingCatalogue.Testing.Data/EntityBuilders/EpicEntityBuilder.cs [hunk 5]
```diff
@@ -8,6 +8,12 @@ public sealed class EpicEntityBuilder
         private readonly EpicEntity _epicEntity;
         private readonly int _compliancyLevelId = 1;
 
+        public enum CompliancyLevel
+        {
+            May = 1,
+            Must= 2
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Testing.Data/EntityBuilders/EpicEntityBuilder.cs]: Spacing?

**Pull request merged**